Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/ui/list/VaccinationListViewModel.kt

```diff
@@ -0,0 +1,203 @@
+                vaccinationDatas = listOf(
+                    VaccinationCertificateV1.VaccinationData(
+                        targetId = "840539006",
+                        vaccineId = "1119349007",
+                        medicalProductId = medicalProductId,
+                        marketAuthorizationHolderId = "ORG-100030215",
+                        doseNumber = 2,
+                        totalSeriesOfDoses = 2,
+                        vaccinatedAt = LocalDate.parse("2021-04-22"),
+                        countryOfVaccination = "NL",
+                        certificateIssuer = "Ministry of Public Health, Welfare and Sport",
+                        uniqueCertificateIdentifier = "urn:uvci:01:NL:THECAKEISALIE",
+                    )
+                )
+            )
+        )
+        val nameDataProofCertificate = ProofCertificateV1.NameData(
+            givenName = "François-Joan",
+            givenNameStandardized = "FRANCOIS<JOAN",
+            familyName = "d'Arsøns - van Halen",
+            familyNameStandardized = "DARSONS<VAN<HALEN",
+        )
+        val proofCertificates = setOf(
+            ProofCertificateV1(
+                version = "1.0.0",
+                nameData = nameDataProofCertificate,
+                dateOfBirth = dateOfBirth,
+                vaccinationDatas = listOf(
+                    ProofCertificateV1.VaccinationData(
+                        targetId = "840539006",
+                        vaccineId = "1119349007",
+                        medicalProductId = medicalProductId,
+                        marketAuthorizationHolderId = "ORG-100030215",
+                        doseNumber = 2,
+                        totalSeriesOfDoses = 2,
+                        vaccinatedAt = LocalDate.parse("2021-04-22"),
+                        countryOfVaccination = "DE",
+                        certificateIssuer = "Ministry of Public Health, Welfare and Sport",
+                        uniqueCertificateIdentifier = "urn:uvci:01:NL:THECAKEISALIE",
+                    )
+                )
+            )
+        )
+
+        val listItems = assembleItemList(
+            vaccinationCertificates = vaccinationCertificates,
+            proofCertificates = proofCertificates,
+            firstName = "François-Joan",
+            lastName = "d'Arsøns - van Halen",
+            dateOfBirth = dateOfBirth,
+            vaccinationStatus,
+            timeStamper.nowUTC.plus(Duration.standardDays(3))
+        )
+
+        UiState(listItems, vaccinationStatus = vaccinationStatus)
+    }.catch {
+        // TODO Error Handling in an upcoming subtask
+    }.asLiveData()
+
+    // TODO: after using actual values from the repository, we only pass VaccinatedPerson here instead of all these
+    // arguments
+    @Suppress("LongParameterList")
+    private fun assembleItemList(
+        vaccinationCertificates: Set<VaccinationCertificateV1>,
+        proofCertificates: Set<ProofCertificateV1>,
```

Comment:
Please work with the interfaces `VaccinationCertificate` and `ProofCertificate`, available via `VaccinatedPerson.vaccinationCertificates/proofCertificates`.

The `Vaccination/ProofCertificateV1` are internal details that should not concern the UI. They also do not contain the translated names for all values (i.e. vaccine name), as these are values retrieved via a server, and have to be resolved to create `VaccinationCertificate` (interface) out of `VaccinationCertificateV1` (raw json mapping)

Reply:
Thanks, changed!